streamparse/dsl/component.py

```diff
@@ -1,30 +1,177 @@
 """
 Component-level Specification
 
-This module is called component to mirror organization of storm package.
+This module is called component to mirror organization of pystorm package.
 """
+from __future__ import absolute_import
 
-from pystorm.component import Component
+from copy import deepcopy
 
+import simplejson as json
+from six import iteritems, string_types
 
-class Specification(object):
-    def __init__(self, component_cls, name=None, parallelism=1):
-        if not issubclass(component_cls, Component):
-            raise TypeError("Invalid component: {}".format(component_cls))
+from .stream import Grouping, Stream
+from .thrift import storm_thrift
+from storm_thrift import (ComponentCommon, ComponentObject, GlobalStreamId,
+                          JavaObject, ShellComponent, StreamInfo)
 
-        if not isinstance(parallelism, int) or parallelism < 1:
+
+class ComponentSpec(object):
+    def __init__(self, component_cls, name=None, inputs=None, parallelism=1,
```

Comment:
Did @rduplain put you up to this? :smile:

He said almost the same thing to me last week.  In general, I despise one-letter arguments.  If you had never used the Clojure DSL, would you guess that `p` meant parallelism?  Is `par` a reasonable halfway point?